From 5b0ea79e6069b2a8f4f99ce8195f21e1c70ddf31 Mon Sep 17 00:00:00 2001
From: Tiago Vignatti <tiago.vignatti@intel.com>
Date: Mon, 15 Jul 2013 17:00:13 -0300
Subject: [PATCH 4/9] ui/aura: Forward GetAcceleratedWidget call to the Ozone
 implementation

It also make sure the size of the window is corresponding with the native
widget implementation (AttemptToResizeAcceleratedWidget).

Signed-off-by: Tiago Vignatti <tiago.vignatti@intel.com>
---
 ui/aura/root_window_host_ozone.cc |   10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/ui/aura/root_window_host_ozone.cc b/ui/aura/root_window_host_ozone.cc
index 55ede01..dde584c 100644
--- a/ui/aura/root_window_host_ozone.cc
+++ b/ui/aura/root_window_host_ozone.cc
@@ -5,6 +5,7 @@
 #include "ui/aura/root_window_host_ozone.h"
 
 #include "ui/aura/root_window.h"
+#include "ui/base/ozone/surface_factory_ozone.h"
 
 namespace aura {
 
@@ -39,9 +40,12 @@ RootWindow* RootWindowHostOzone::GetRootWindow() {
 }
 
 gfx::AcceleratedWidget RootWindowHostOzone::GetAcceleratedWidget() {
-  // TODO(rjkroege): Support more than one screen. Screen ids start at 1 and
-  // increase for each screen.
-  return (gfx::AcceleratedWidget)1;
+  gfx::AcceleratedWidget w =
+    ui::SurfaceFactoryOzone::GetInstance()->GetAcceleratedWidget();
+
+  ui::SurfaceFactoryOzone::GetInstance()->AttemptToResizeAcceleratedWidget(w, bounds_);
+
+  return w;
 }
 
 void RootWindowHostOzone::Show() { NOTIMPLEMENTED(); }
-- 
1.7.9.5

